tf2_ros/include/tf2_ros/transform_listener.h

```diff
@@ -52,6 +52,8 @@ class TransformListener
   /**@brief Constructor for transform listener */
   TransformListener(tf2::BufferCore& buffer, bool spin_thread = true);
   TransformListener(tf2::BufferCore& buffer, const ros::NodeHandle& nh, bool spin_thread = true);
+  TransformListener(tf2::BufferCore& buffer, ros::TransportHints transport_hints, bool spin_thread = true);
```

Comment:
This is a pretty advanced feature, and I don't think that we need to support the combinatoric of both a node handle and non-nodehandle version of this API.
We might as well add it as one more optional/default argument appended to the longer signature.

Reply:
I added an additional default argument initially, but I found that it was breaking ABI and would force me to build some other repos from source (in particular geometry/tf package).
I thought additional constructors would have less side effects and get the PR accepted more easily :)

But if that is your preferred approach I can go back to it.
In that case I would add one more optional/default argument to both constructors.